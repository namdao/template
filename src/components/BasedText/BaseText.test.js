import React from 'react';
import { shallow } from 'enzyme';
import BaseText from './index';

const setup = (propOverrides) => {
  const props = {
    children: 'Test',
    ...propOverrides,
  };
  const wrapper = shallow(<BaseText {...props} />);
  return {
    props,
    wrapper,
  };
};

describe('BaseText check snapShot', () => {
  test('should match snapshot', () => {
    const { wrapper } = setup();
    expect(wrapper).toMatchSnapshot();
  });
});

describe('BaseText check render', () => {
  test('should render with title text', () => {
    const { wrapper } = setup();
    expect(wrapper.find('Text').props().children).toEqual('Test');
  });
  test('should can pass props color', () => {
    const { wrapper } = setup({
      color: 'red',
      size: 1,
      weight: '500',
    });
    expect(wrapper.find('Text')).toMatchSnapshot();
  });
});
