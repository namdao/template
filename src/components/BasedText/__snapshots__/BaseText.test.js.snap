// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseText check render should can pass props color 1`] = `
<Text
  style={
    Array [
      Object {},
      Object {
        "color": "red",
        "fontSize": 1,
        "fontWeight": "500",
        "lineHeight": undefined,
        "textAlign": "auto",
      },
    ]
  }
>
  Test
</Text>
`;

exports[`BaseText check snapShot should match snapshot 1`] = `
<Text
  style={
    Array [
      Object {},
      Object {
        "color": null,
        "fontSize": null,
        "fontWeight": null,
        "lineHeight": undefined,
        "textAlign": "auto",
      },
    ]
  }
>
  Test
</Text>
`;
